Classes/Decoder/UrlDecoder.php

```diff
@@ -1162,16 +1162,28 @@ protected function makeRealPhpArrayFromRequestVars(array $requestVariables) {
 	 * not set already.
 	 */
 	protected function mergeGetVarsFromDomainsConfiguration() {
-		$getVars = array();
 		// Convert the configuration into an $_GET-"friendly" format
-		// Maybe there's an nicer way to do this
-		$getVarsAsQueryString = http_build_query($this->configuration->getGetVarsToSet());
-		parse_str($getVarsAsQueryString, $getVars);
-
-		// Over-write $_GET-params to match earlier behavior …
-		ArrayUtility::mergeRecursiveWithOverrule($getVars, $_GET, TRUE, TRUE, FALSE);
-		// … and store it back
-		$_GET = $getVars;
+		$getVars = $this->makeRealPhpArrayFromRequestVars($this->configuration->getGetVarsToSet());
+
+		// Over-write with $_GET-params that $_GET-parmas have a "higher" priority
+		$GET = GeneralUtility::_GET();
+		if (!is_array($GET)) {
+			$GET = array();
+		}
+		ArrayUtility::mergeRecursiveWithOverrule($getVars, $GET, TRUE, TRUE, FALSE);
+
+		// Re-caclulate the cHash
```

Comment:
cHash will be correct because the parameter removed by the encoding process and re-created by the decoding process.